### PR TITLE
Use features in search

### PR DESCRIPTION
**Pull request by lhoestq**
The search endpoint was using the feature types from the arrow table returned by duckdb, which doesn't contain any metadata about the Image type.

So I added a `features` field to the `split-duckdb-index` job to store the feature types that the search endpoint can use to correctly load the image data.

I added a migration job to fill this field with `None` and I will simply update the `split-duckdb-index` job for image datasets.

Fix https://github.com/huggingface/datasets-server/issues/1713

### CODE REVIEW

File: jobs/mongodb_migration/src/mongodb_migration/migrations/_20230824154999_cache_add_features_field_in_split_duckdb_index.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
```
Comment (AndreaFrancis) [inline jobs/mongodb_migration/src/mongodb_migration/migrations/_20230824154999_cache_add_features_field_in_split_duckdb_index.py:2]:
```suggestion
# Copyright 2023 The HuggingFace Authors.
```

File: jobs/mongodb_migration/src/mongodb_migration/migrations/_20230824154999_cache_add_features_field_in_split_duckdb_index.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from libcommon.constants import CACHE_COLLECTION_RESPONSES, CACHE_MONGOENGINE_ALIAS
+from libcommon.simple_cache import CachedResponseDocument
+from mongoengine.connection import get_db
+
+from mongodb_migration.check import check_documents
+from mongodb_migration.migration import Migration
+
+
+# connection already occurred in the main.py (caveat: we use globals)
+class MigrationAddFeaturesToSplitDuckdbIndexCacheResponse(Migration):
```
Comment (AndreaFrancis) [inline jobs/mongodb_migration/src/mongodb_migration/migrations/_20230824154999_cache_add_features_field_in_split_duckdb_index.py:15]: Not critical: Can you add a test for this migration? To ensure the new field is added

**Review by AndreaFrancis — APPROVED** [review]
Thanks!

**Review by severo — APPROVED** [review]
Excellent. Can you update the openapi spec as well? (it's in docs/source)

**Comment by lhoestq** [comment]
Will update the spec with a real example once it's deployed

**Comment by lhoestq** [comment]
Just deleted the cached responses of datasets with an `"image": Image()` type to refresh them (around 13k documents).

Not sure how to get the list of datasets with an `Image()` type in a column named differently than `image` though without scanning all the db

**Comment by severo** [comment]
> Not sure how to get the list of datasets with an Image() type in a column named differently than image though without scanning all the db

Do you have an idea @AndreaFrancis 

> Will update the spec with a real example once it's deployed

Could you do it?

**Comment by severo** [comment]
> (fixed by freeing some inodes)
cached-assets now use an EFS volume instead of the limited NFS, so this issue should disappear for some time. Now, only the assets (/first-rows) are stored in the NFS.

**Comment by severo** [comment]
> Will update the spec with a real example once it's deployed

https://github.com/huggingface/datasets-server/pull/1787

**Pull request merged**